x-pack/plugins/snapshot_restore/public/application/components/policy_form/steps/step_settings/fields/include_feature_states_field/include_feature_states_field.tsx

```diff
@@ -0,0 +1,125 @@
+ * 2.0.
+ */
+
+import React, { FunctionComponent, useMemo } from 'react';
+import { FormattedMessage } from '@kbn/i18n-react';
+
+import {
+  EuiDescribedFormGroup,
+  EuiFormRow,
+  EuiSwitch,
+  EuiSwitchEvent,
+  EuiTitle,
+  EuiCallOut,
+  EuiSpacer,
+  EuiComboBoxOptionOption,
+} from '@elastic/eui';
+
+import { FEATURE_STATES_NONE_OPTION } from '../../../../../../../../common/constants';
+import { SlmPolicyPayload } from '../../../../../../../../common/types';
+import { PolicyValidation } from '../../../../../../services/validation';
+import { useLoadFeatures } from '../../../../../../services/http/policy_requests';
+import { FeatureStatesFormField } from '../../../../../feature_states_form_field';
+
+interface Props {
+  policy: SlmPolicyPayload;
+  onUpdate: (arg: Partial<SlmPolicyPayload['config']>) => void;
+  errors: PolicyValidation['errors'];
+}
+
+export type FeaturesOption = EuiComboBoxOptionOption<string>;
+
+export const IncludeFeatureStatesField: FunctionComponent<Props> = ({ policy, onUpdate }) => {
+  const { config = {} } = policy;
+  const {
+    error: errorLoadingFeatures,
+    isLoading: isLoadingFeatures,
+    data: featuresResponse,
+  } = useLoadFeatures();
+
+  const featuresOptions = useMemo(() => {
+    const features = featuresResponse?.features || [];
+    return features.map((feature) => feature.name);
+  }, [featuresResponse]);
+
+  const selectedOptions = useMemo(() => {
+    return config?.featureStates?.map((feature) => ({ label: feature })) as FeaturesOption[];
+  }, [config.featureStates]);
+
+  const isFeatureStatesToggleEnabled =
+    config.featureStates !== undefined &&
+    !config.featureStates.includes(FEATURE_STATES_NONE_OPTION);
+
+  const onFeatureStatesToggleChange = (event: EuiSwitchEvent) => {
+    const { checked } = event.target;
+
+    onUpdate({
+      featureStates: checked ? [] : [FEATURE_STATES_NONE_OPTION],
+    });
+  };
+
+  return (
+    <EuiDescribedFormGroup
+      title={
+        <EuiTitle size="s">
+          <h3>
+            <FormattedMessage
+              id="xpack.snapshotRestore.policyForm.stepSettings.includeFeatureStatesDescriptionTitle"
+              defaultMessage="Include feature state"
+            />
+          </h3>
+        </EuiTitle>
+      }
+      description={
+        <FormattedMessage
+          id="xpack.snapshotRestore.policyForm.stepSettings.includeFeatureStatesDescription"
+          defaultMessage="Stores the configuration, history, and other data stored in Elasticsearch by a feature such as Elasticsearch security."
+        />
+      }
+      fullWidth
+    >
+      <EuiFormRow hasEmptyLabelSpace fullWidth>
+        <EuiSwitch
+          data-test-subj="featureStatesToggle"
+          label={
+            <FormattedMessage
+              id="xpack.snapshotRestore.policyForm.stepSettings.policyIncludeFeatureStatesLabel"
+              defaultMessage="Include feature state from"
```

Comment:
I ran this by @gchaps & she had a great question: Do we need both the toggle & the menu? It's not a pattern that we could find used elsewhere--it's either an on/off toggle, or a menu with a heading. She suggested using "Features to include" for the heading, and then just having the drop down with the options: 

All features
No features
<feature1>
<feature2>
...
<featureN>

Reply:
We use a somehow similar pattern in the `Data streams and indices` section, in which when toggled off we show a list of indices and datastreams for the user to select. I think it will feel less visually consistent if we use a just a dropdown to represent all, none, or multiple choices. Not sure what you think about this @cristina-eleonora?

Reply:
We will address this in a separate PR, I've created an issue for this https://github.com/elastic/kibana/issues/132211